zingolib/src/wallet/keys/extended_transparent.rs

```diff
@@ -102,9 +102,9 @@ impl ExtendedPrivKey {
                 KeyIndex::hardened_from_normalize_index(config.get_coin_type()).unwrap(),
             )
             .unwrap()
-            .derive_private_key(KeyIndex::hardened_from_normalize_index(0).unwrap())
+            .derive_private_key(KeyIndex::hardened_from_normalize_index(position).unwrap())
             .unwrap()
-            .derive_private_key(KeyIndex::Normal(position))
+            .derive_private_key(KeyIndex::Normal(0))
             .unwrap()
```

Comment:
The BIP-44 rules (which Zcash espouses for transparent addresses) prescribe a derivation path of
`m / purpose' / coin_type' / account' / change / address_index`
After this fix, this derivation gets this far:
`m / purpose' / coin_type' / account' / change`

Where does the `address_index` derivation happen? It needs to happen somewhere (even if the address_index is always 0) or the derived address won't match what other wallets derive.

Reply:
It looks like the `new_address` function in zingolib does indeed derive another level, but I'm not certain. If so, then what's here is standards-compliant to the best of my knowledge.
Further, it looks like `new_address` creates a unique transparent address every time it's called. That's cool, so long as the wallet file remembers that all these addresses have been created so that they are all individually checked for UTXOs forever after. Technically, it violates the [address gap limit rule](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#user-content-Address_gap_limit) in BIP-44 because it could generate more than 20 unique transparent addresses without any of them having actually received any funds, which makes restoring a wallet from seed likely to never discover the funds at an address beyond 20 addresses that have not been used.

Reply:
> The BIP-44 rules (which Zcash espouses for transparent addresses) prescribe a derivation path of `m / purpose' / coin_type' / account' / change / address_index` After this fix, this derivation gets this far: `m / purpose' / coin_type' / account' / change`

Before the fix, it was incorrectly deriving `m / purpose' / coin_type' / change' / account`, which doesn't break anything for our use-case because `account` and `change` are both always zero.
> Where does the `address_index` derivation happen? It needs to happen somewhere (even if the address_index is always 0) or the derived address won't match what other wallets derive.

The `address index` derives a new level, yes.